src/core_modules/capture-core/components/Pages/EnrollmentEditEvent/EnrollmentEditEventPage.container.js

```diff
@@ -147,6 +154,20 @@ const EnrollmentEditEventPageWithContextPlain = ({
         programStage,
         event,
     });
+    const assignee = useAssignee(event);
+    const onGetAssignedUserSaveContext = useAssignedUserSaveContext(event);
```

Comment:
That is a good question. The callback is useful to ensure that we only calculate the data we need to send to the server when we assign a user (when the "Save" button is clicked). In the case of [EnrollmentEditEventPage.container](https://github.com/dhis2/capture-app/pull/3412/files#diff-eeac4f86e764871045b77688d009436440c3f98f09af8a1c2b7d18b44d642105R158), the event information is already there, so the callback brings no value. This is not the same for [ViewEvent.container.js](https://github.com/dhis2/capture-app/pull/3412/files#diff-570dde8a388207f6f3a7ad8885ccfa2b4fa6db15b03f6d4f2fc1e6fc71d06622R24), where we calculate and convert the event when assigning a user. Because `WidgetAssigne` has the same interface for both pages the callback is used in both. Do you have other suggestions on how to handle this? Thanks!

 I agree that the 'on'- prefix is a bit off and I removed it.

Reply:
We could in theory call `getAssignedUserSaveContext` outside `WidgetAssignee` and pass the result as a prop. In practice there is no difference though, the only thing we gain by changing this is to make the code a little less convoluted and make this line a little less puzzling. So maybe not worth the effort?

Reply:
> The callback is useful to ensure that we only calculate the data we need to send to the server when we assign a user (when the "Save" button is clicked).
Ah, delayed calculation... That favours the current solution. Let's just keep it the way it currently is then :+1: